src/map.rs

```diff
@@ -380,17 +380,23 @@ where
                     let mut p: Option<&Node<K, V>> = Some(node);
                     while p.is_some() {
                         delta -= 1;
+                        unsafe {
+                            guard.defer_destroy(p.unwrap().value.load(Ordering::SeqCst, guard));
+                        }
                         p = match p.unwrap().next.load(Ordering::SeqCst, guard) {
                             s if s.is_null() => None,
                             s => Some(
                                 unsafe { s.deref() }
                                     .as_node()
                                     .expect("Node.next should always be BinEntry::Node"),
                             ),
-                        }
+                        };
                     }
                     tab.store_bin(idx, Shared::null());
                     idx += 1;
+                    unsafe {
```

Comment:
This will only free the _first_ node, but _all_ the nodes need to be freed.
Also, this should include a safety argument -- see `remove`.

Reply:
But `raw_node' is the node to which the current index points which increases every loop cycle. Doesn't this mean that all nodes are destroyed?

Reply:
Ah, but remember that every bin is a linked list. `raw_node` is the _first_ node in the linked list in bin `idx`, but you also need to free all the nodes along the `.next` pointers (that's what the `while` loop just above walks).

Reply:
Ah okay, makes sense.